### PR TITLE
Crop Cloud Cropped Word Movement

### PR DESCRIPTION
**Pull request by peterger8y**
# Description

This pull request implements basic cropped word movements demonstrated to stakeholders. Word movements include simple bungie like movements, wiggles, and spinning. All changes isolated to crop_cloud.py -> look no further than app/utils/visualizations/crop_cloud.py for changes.

loom video for brief synopsis:
https://www.loom.com/share/c17ab7aeed764ff1bdf84530d197f82e

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

## Has This Been Tested

- [x] Yes
- [ ] No
- [ ] Not necessary

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

File: app/utils/visualizations/crop_cloud.py [hunk 1]
```diff
@@ -39,11 +47,19 @@ def load_image(filename, max_length=None):
     return image
 
 
-# Convert the image into black and white, separating the text from the background
 def make_monochrome(image, blur=1, block_size=31, c=13):
-    # blur must be an odd number >= 1
-    # block_size is for smoothing out a varying exposure. too small etches out text. must be an odd number > 1
-    # c is for denoising. too small and you have noise. too big erodes text.
+    """
```
Comment (BritVandy) [inline app/utils/visualizations/crop_cloud.py:51]: Thank you for adding more documentation for the functions.

File: app/utils/visualizations/crop_cloud.py [hunk 2]
```diff
@@ -246,26 +328,304 @@ def pick_x(canvas_width, word_width, date_number=None, total_dates=None):
...
+    Input:
+        `arr` array of single x and y coordinate
+        `frame` frame number for positional reference
+        `pic_width` image.shape[1]
+        `pic_height` image.shape[0]
+
+    Output:
+        returns pixel adjustment for y coordinate
+    """
+    frame = ((2 * np.pi) / 12) * frame
```
Comment (BritVandy) [inline app/utils/visualizations/crop_cloud.py:396]: It’s cool that you were able to accomplish movement with basically just math.

File: app/utils/visualizations/crop_cloud.py [hunk 3]
```diff
@@ -246,26 +328,304 @@ def pick_x(canvas_width, word_width, date_number=None, total_dates=None):
...
+                        canvas_to_app[coord_ind1][coord_ind2] = image[coord[0]][coord[1]] * coord[2] + (
+                                (1 - coord[2]) * canvas_to_app[coord_ind1][coord_ind2])
+            elif move_type_ind == 2:
+                for coord in pos[0]:
+                    coords = spinner(coord[:2], frame, image)
+                    coord_ind2 = int(coords[1] + x1)
+                    coord_ind1 = int(coords[0] + y1)
+                    if 0 < coord_ind1 < canvas_to_app.shape[0] and 0 < coord_ind2 < canvas_to_app.shape[1]:
+                        canvas_to_app[coord_ind1][coord_ind2] = image[coord[0]][coord[1]] * coord[2] + (
+                                (1 - coord[2]) * canvas_to_app[coord_ind1][coord_ind2])
```
Comment (BritVandy) [inline app/utils/visualizations/crop_cloud.py:584]: You have move_type_ind == 1 twice. In the if and the last elif. Not sure if that’s supposed to be like that or not.

**Review by BritVandy — APPROVED** [review]
Overall this looks good. So glad the stakeholders loved the animation you did! Way to go!!

**Review by Fellylove — APPROVED** [review]
Great job, Peter!

You've done a good job of explaining how the code works. I think the next cohort will find the video useful.

**Review by techthumb1 — APPROVED** [review]
Really great work Peter! Code is clean and implemented nicely, and well documented. I feel the changes you incorporated improved the model, good job!

**Pull request merged**